tests/shared/src/test/scala/munit/CustomPrinterSuite.scala

```diff
@@ -0,0 +1,141 @@
+package munit
+
+import org.scalacheck.Prop.forAll
+import munit.internal.console.Printers
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Prop
+
+class CustomPrinterSuite extends FunSuite with ScalaCheckSuite {
+
+  private case class Foo(i: Int)
+
+  private case class Bar(l: List[Int])
+
+  private case class FooBar(foo: Foo, bar: Bar)
+
+  private val genFoo = arbitrary[Int].map(Foo(_))
+
+  // limit size to 10 to have a reasonable number of values
+  private val genBar = arbitrary[List[Int]].map(l => Bar(l.take(10)))
+
+  private val genFooBar = for {
+    foo <- genFoo
+    bar <- genBar
+  } yield FooBar(foo, bar)
+
+  private val longPrinter: Printer = Printer.apply { case l: Long =>
+    s"MoreThanInt($l)"
+  }
+
+  private val fooPrinter: Printer = Printer.apply { case Foo(i) =>
+    s"Foo(INT($i))"
+  }
+
+  private val listPrinter: Printer = Printer.apply { case l: List[_] =>
+    l.mkString("[", ",", "]")
+  }
+
+  private val intPrinter: Printer = Printer.apply { case i: Int =>
+    s"NotNaN($i)"
+  }
+
+  private val isScala213: Boolean = BuildInfo.scalaVersion.startsWith("2.13")
+
+  private def checkProp(
+      options: TestOptions,
+      isEnabled: Boolean = true
+  )(p: => Prop): Unit = {
+    test(options) {
+      assume(isEnabled, "disabled test")
+      p
+    }
+  }
+
+  checkProp("long") {
+    forAll(arbitrary[Long]) { (l: Long) =>
+      val obtained = Printers.print(l, longPrinter)
+      val expected = s"MoreThanInt($l)"
+      assertEquals(obtained, expected)
+    }
+  }
+
+  checkProp("list") {
+    forAll(arbitrary[List[Int]]) { l =>
+      val obtained = Printers.print(l, listPrinter)
+      val expected = l.mkString("[", ",", "]")
+      assertEquals(obtained, expected)
+    }
+  }
+
+  checkProp("product") {
+    forAll(genFoo) { foo =>
+      val obtained = Printers.print(foo, fooPrinter)
+      val expected = s"Foo(INT(${foo.i}))"
+      assertEquals(obtained, expected)
+    }
+  }
+
+  checkProp("int in product", isEnabled = isScala213) {
+    forAll(genFoo) { foo =>
+      val obtained = Printers.print(foo, intPrinter)
+      val expected = s"Foo(\n  i = NotNaN(${foo.i})\n)"
+      assertEquals(obtained, expected)
+    }
+  }
+
+  checkProp("list in product", isEnabled = isScala213) {
+    forAll(genBar) { bar =>
+      val obtained = Printers.print(bar, listPrinter)
+      val expected = s"Bar(\n  l = ${bar.l.mkString("[", ",", "]")}\n)"
+      assertEquals(obtained, expected)
+    }
+  }
+
+  checkProp("list and int in product", isEnabled = isScala213) {
+    forAll(genFooBar) { foobar =>
+      val obtained = Printers
+        .print(foobar, listPrinter.orElse(intPrinter))
+        .stripMargin
+        .stripLeading()
```

Comment:
Looks like `.stripLeading()` was added in JDK 11, so this will not compile currently.

Reply:
![image](https://user-images.githubusercontent.com/55251330/228490743-f3c18ee0-1f90-4476-b218-56e2565bcbe2.png)

Fixed ! This made the updated test a bit less specific but should be acceptable since more thorough checks on whitespaces, indents etc. are made elsewhere